conference.js

```diff
@@ -1664,27 +1664,47 @@ export default {
             return;
         }
 
+        // Create a new presenter track and apply the presenter effect.
         if (!this.localPresenterVideo && !mute) {
-            // create a new presenter track and apply the presenter effect.
             let { height } = this.localVideo.track.getSettings();
-
-            // Workaround for Firefox since it doesn't return the correct width/height of the desktop stream
-            // that is being currently shared.
-            if (!height) {
-                const desktopResizeConstraints = {
+            const { aspectRatio, width } = this.localVideo.track.getSettings();
+            let desktopResizeConstraints = {};
+            let resizeDesktopStream = false;
+
+            // Determine the constraints if the desktop track needs to be resized.
+            // Resizing is needed when the resolution cannot be determined or when
+            // the window is bigger than 1280x720.
+            if (height && width && aspectRatio) {
```

Comment:
Is there a device which doesn't expose the aspecct ratio? I guess we should be able to figure it out if we have height and width, right?

Reply:
yes we can calculate aspect ratio from height and width, will remove the check for aspect ratio here.